source/module_relax/relax_old/ions_move_bfgs.cpp

```diff
@@ -40,7 +40,16 @@ void Ions_Move_BFGS::start(UnitCell& ucell, const ModuleBase::matrix& force, con
     // istep must be set eariler.
 
     // use force to setup gradient.
-    Ions_Move_Basic::setup_gradient(ucell, force, this->pos, this->grad);
+    if (first_step)
+    {
+        Ions_Move_Basic::setup_gradient(ucell, force, this->pos, this->grad);
+        first_step = false;
+    }
+    else{
+        double *pos_tmp = new double[3 * ucell.nat];
```

Comment:
Maybe I suggest use std::vector here because now it is the time we can gradually get rid of the unsafe raw pointer...